pymatgen/analysis/reaction_rates.py

```diff
@@ -0,0 +1,566 @@
+        Calculate entropy of activation.
+
+        Args:
+            reverse (bool): if True (default False), consider the reverse reaction; otherwise,
+                consider the forwards reaction
+
+        Returns:
+            float: entropy of activation (in cal/mol-K)
+
+        """
+
+        trans_entropy = self.transition_state.entropy
+
+        if reverse:
+            pro_entropies = [p.entropy for p in self.products]
+            return trans_entropy - sum(pro_entropies)
+        else:
+            rct_entropies = [r.entropy for r in self.reactants]
+            return trans_entropy - sum(rct_entropies)
+
+    def calculate_act_gibbs(self, temperature, reverse=False):
+        """
+        Calculate Gibbs free energy of activation at a given temperature.
+
+        ΔG = ΔH - T ΔS
+
+        Args:
+            temperature (float): absolute temperature in Kelvin
+            reverse (bool): if True (default False), consider the reverse reaction; otherwise,
+                consider the forwards reaction
+
+        Returns:
+            float: Gibbs free energy of activation (in kcal/mol)
+        """
+
+        return self.calculate_act_enthalpy(reverse=reverse) - temperature * self.calculate_act_entropy(reverse=reverse) / 1000
+
+    def calculate_act_thermo(self, temperature=300.0, reverse=False):
+        """
+        Calculate thermodynamics of activation for the reaction.
+
+        Args:
+            temperature (float): absolute temperature in Kelvin (default 300.0K)
+            reverse (bool): if True (default False), consider the reverse reaction; otherwise,
+                consider the forwards reaction
+
+        Returns:
+            act_thermo: dict with relevant activation thermodynamic variables
+        """
+
+        act_thermo = {"energy": self.calculate_act_energy(reverse=reverse),
+                      "enthalpy": self.calculate_act_enthalpy(reverse=reverse),
+                      "entropy": self.calculate_act_entropy(reverse=reverse),
+                      "gibbs": self.calculate_act_gibbs(temperature, reverse=reverse)}
+
+        return act_thermo
+
+    def calculate_rate_constant(self, temperature=300.0, reverse=False, kappa=1.0):
+        """
+        Calculate the rate constant k by the Eyring-Polanyi equation of transition state theory.
+
+        Args:
+            temperature (float): absolute temperature in Kelvin
+            reverse (bool): if True (default False), consider the reverse reaction; otherwise,
+                consider the forwards reaction
+            kappa (float): transmission coefficient (by default, we assume the assumptions of
+                transition-state theory are valid, so kappa = 1.0
+
+        Returns:
+            k_rate (float): temperature-dependent rate constant
+        """
+
+        gibbs = self.calculate_act_gibbs(temperature=temperature, reverse=reverse)
+
+        k_rate = kappa * k * temperature / h * np.exp(-gibbs * 4184 / (R * temperature))
```

Comment:
Where did this 4184 come from?

Reply:
Converting Gibbs from kcal/mol to J/mol (4184 J/kcal).